docs/modules/cluster-performance/pages/thread-per-core-tpc.adoc

```diff
@@ -0,0 +1,97 @@
+= Thread-Per-Core (TPC)
+:description: Find out more about Thread-Per-Core (TPC) and how to enable this feature on clients and cluster members.
+:page-enterprise: true
+
+Enable Thread-Per-Core (TPC) on both clients and cluster members to improve system performance, especially on machines with either fewer or more CPU cores than the recommended hardware listed on the xref:ROOT:production-checklist.adoc[].  
+
+[[tpc-what]]
+== What is Thread-Per-Core (TPC)?
+
+The threading model is responsible for networking, storage and compute, which means it has a big impact on system performance. With Staged Event-Driven Architecture (SEDA) you have dedicated threads that do one particular task (for example, reading from a socket or evaluating a map.get) and thread pools that manage these threads.  
+
+SEDA can lead to throughput and latency issues on machines with limited cores (fewer than 8), as the frequent context switching and state sharing can add significant overhead. Conversely, on machines with a very high number of CPUs (more than 32), a bottleneck can be created by the limited number of I/O threads vs. operation threads. Typically, smaller servers are more readily available through virtualization and for cloud-specific workloads, which can limit system performance in a SEDA environment. 
+
+A Thread-Per-Core (TPC) design uses one thread for every CPU core and every thread does everything. With each TPC thread able to do networking/storage/compute, scaling is mostly a case of increasing the number of threads to spread the work. This means you can scale up and down easily based on the number of available cores, and make more efficient use of resources on machines with either a low or high number of cores. Every task is non-blocking and you can scale vertically much easier, which means high throughput, low latency and significantly improved system performance.  
```

Comment:
```suggestion
A Thread-Per-Core (TPC) design uses one thread for every CPU core and every thread does everything. With each TPC thread able to do networking/storage/compute, scaling is mostly a case of increasing the number of threads to spread the work. This means you can scale up and down based on the number of available cores, and make more efficient use of resources on machines with either a low or high number of cores. Every task is non-blocking and you can scale vertically smoothly, which means high throughput, low latency and significantly improved system performance.  
```

Reply:
I just don't like telling the user something is easy/easier - that might not be their perception ;-)

Reply:
Hmm, I know what you mean but this more a case of the system finding it easier rather than anything you/a human does, which is not reflected by my "you can scale" comment admittedly. Will amend the text...